src/posthog-core.ts

```diff
@@ -2154,9 +2156,34 @@ export class PostHog {
     }
 
     _captureInitialPageview(): void {
-        if (document && !this._initialPageviewCaptured) {
+        if (!document) {
+            return
+        }
+
+        // If page is not visible, add a listener to detect when the page becomes visible
+        // and trigger the pageview only then
+        // This is useful to avoid `prerender` calls from Chrome/Wordpress/SPAs
+        // that are not visible to the user
+
+        if (document.visibilityState !== 'visible') {
```

Comment:
much prefer this check btw
very conflicting information out there about whether `preprender` is a valid value, it's certainly not in the spec

(i wonder if we should have visibility state by default on events)

Reply:
Yeah, types actually fail if I try using `prerender`, so this is definitely the right way to do